layouts/default.vue

```diff
@@ -1,3 +1,34 @@
+<script setup lang="ts">
+type Theme = 'light' | 'dark'
+
+interface Message {
+  type: 'theme'
+  theme: Theme
+}
+
+function setTheme(theme: Theme) {
+  document.documentElement.classList.remove('light', 'dark')
+  document.documentElement.classList.add(theme)
+}
+
+const { initialTheme } = useRoute().query
+
+onMounted(() => {
+  if (initialTheme && (initialTheme === 'light' || initialTheme === 'dark')) {
+    setTheme(initialTheme)
+  }
+
+  window.addEventListener('message', (event: MessageEvent<Message>) => {
```

Comment:
Consider using [useBroadcastChannel](https://vueuse.org/core/useBroadcastChannel/#usebroadcastchannel) instead.
Ideally extracted in `composables/useBroadcastChannelWithIframeHost.ts` or whichever better name.

Reply:
TIL cool API!

Unfortunately it seems it won't work for our use case.
The Broadcast API only works on pages on with the same [origin](https://developer.mozilla.org/en-US/docs/Glossary/Origin). (aka having different ports/subdomains won't do)

Reply:
That's how I envisioned the final architecture and why I mentioned using localStorage initially and [having the host app serving the build artifact itself in the related PR](https://github.com/maninak/radicle-interface-with-planning-boards/pull/1#issuecomment-1849072801)

I'm gonna leave the final call for using `Local Storage`, `Window.postMessage()` or `Broadcast API` up to you.

Reply:
I'm going to go with `Window.postMessage` over `LocalStorage` because it's specifically built for communication between windows.

Under different circumstances I would have chosen `Broadcast API`, but because the time setting up the projects to enable is IMO better spent on other features, I won't.